docs/extending-and-experimenting.md

```diff
@@ -147,3 +147,130 @@ subscribe to the result of `executeQuery`, keep track
+
+> Note: The easiest way to uniquely identify an operation is by checking it's `key` property. This is a uniquely generated hash intended for this purpose.
+
+```ts
+import { filter, pipe, tap } from 'wonka';
+import { Exchange } from 'urql';
+
+export const dedupExchange: Exchange = ({ forward }) => {
+  const inFlight = new Set<string>();
+
+  return ops$ => {
+    const preFlight$ = pipe(
+      ops$,
+      // "tap" is called every time a new operation comes through.
+      // "tap" is invisible to subsequent stream operations
+      tap(operation => {
+        if (operation.operationName !== 'query') {
+          return;
+        }
+
+        inFlight.add(operation.key);
+      })
+    );
+
+    return pipe(
+      forward(preFlight$),
+      // This `tap` is called after an operation has been completed
+      tap(response => inFlight.delete(response.operation.key))
+    );
+  };
+};
+```
+
+Our exchange now tracks operations and their completion, but for this to be useful,
+we need to stop in flight operations being forwarded to the next exchange. In order to do this,
+we're now going to have to use the `filter` function. To simplify things further, we can group all our pre-flight logic together.
+
+```ts
+import { filter, pipe, tap } from 'wonka';
+import { Exchange } from 'urql';
+
+export const dedupExchange: Exchange = ({ forward }) => {
+  const inFlight = new Set<string>();
+
+  return ops$ => {
+    const preFlight$ = pipe(
+      ops$,
+      filter(operation => {
+        // Don't filter non-query operations
+        if (operation.operationName !== 'query') {
+          return true;
+        }
+
+        const isInFlight = inFlight.has(operation.key);
+        inFlight.add(operation.key);
+
+        return isInFlight;
+      })
+    );
+
+    return pipe(
+      forward(preFlight$),
+      // This `tap` is called after an operation has been completed
+      tap(response => inFlight.delete(response.operation.key))
+    );
+  };
+};
+```
+
+Hopefully you've been following this example along, if that's the case, congratulations! You just created your first exchange. The only thing left is to add it to `urql` on client creation.
+
+> Note: Remember, operations are forwarded in a linear fashion. Depending on your exchange, ordering may be important (in this example, dedup needs to come before fetch otherwise it wouldn't do anything).
+
+```ts
+import {
+  createClient,
+  cacheExchange,
+  debugExchange,
+  fetchExchange,
+} from 'urql';
+import { dedupExchange } from './myDedupExchange';
+
+export const urqlClient = createClient({
+  url: 'https://localhost:3000/graphql',
+  exchanges: [dedupExchange, debugExchange, cacheExchange, fetchExchange],
+});
+```
+
+To conclude, here's a quick recap of what we've achieved:
```

Comment:
Lets ret rid of the bullet point list summarising sole things :) it seems out of place given the short length of this section